jbehave-support-core/src/main/java/org/jbehavesupport/core/rest/RestServiceHandler.java

```diff
@@ -125,6 +127,43 @@ public void sendRequest(String urlPath, HttpMethod requestMethod, ExamplesTable
         } catch (HttpClientErrorException | HttpServerErrorException e) {
             storeResponse(e);
         }
+        isSuccessful();
+    }
+
+    private void isSuccessful() {
+        if (testContext.keySet().containsAll(
+            Arrays.asList(REST_RESPONSE_CODE, REST_RESPONSE_HEADERS, REST_RESPONSE_JSON))) {
+            isSuccessful(HttpStatus.valueOf((Integer)testContext.get(REST_RESPONSE_CODE)),
+                testContext.get(REST_RESPONSE_HEADERS),
+                testContext.get(REST_RESPONSE_JSON));
+        } else {
+            log.warn("Test context doesn't have consistent REST response stored, skipping success handler");
```

Comment:
Is this situation ever possible? It seems to me that at the end of sendRequest method, there are always those values in TestContext.

Reply:
It really shouldn't happen, but I want to avoid situations where user receives information "some key is not present in test context" while calling some API.